.github/workflows/cancel-dupliate-workflow-runs.yml

```diff
@@ -0,0 +1,39 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+name: "Cancel Duplicates"
+on:
+  workflow_run:
+    workflows: 
+      - 'Java CI'
+      - 'Python CI'
+    types: ['requested']
+
+jobs:
+  cancel-duplicate-workflow-runs:
+    name: "Cancel duplicate workflow runs"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: potiuk/cancel-workflow-runs@4723494a065d162f8e9efd071b98e0126e00f866 # @master
```

Comment:
Could you add a comment that explains why a specific commit hash is used? I think the reason is to avoid picking up changes to the workflow because while we can look at what a specific commit is doing, we don't want to run whatever happens to be in someone else's repository right?

Reply:
That is why I added that tag. Additionally, the Spark repo pins as well. Happy to add a comment, but see my other note about potentially using the "beta" Github Actions feature called `concrurrency` instead (which avoids the need for a token).

Reply:
Done (it's in the lines above this).